### DISCUSSION TITLE
Azure DevOps (XCode Project) iOS export fails

### DISCUSSION THREAD

**Discussion by syedimranFB** [community/community#55634] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi, we are working with ionic/cordova project with xcode in azure devops, we have installed all the necessary certificates and mobile provisioning profile on our azure devops build agent, so far we are able to build and archive, when we are opting for an archive task it is throwing 

```
(Error Domain=IDEProvisioningErrorDomain Code=25 "There are no accounts registered with Xcode." UserInfo={IDEDistributionIssueSeverity=3, NSLocalizedDescription=There are no accounts registered with Xcode., NSLocalizedRecoverySuggestion=Add your developer account to Xcode}

error: exportArchive: No profiles for 'com.timer.chronologger' were found

Error Domain=IDEProfileLocatorErrorDomain Code=1 "No profiles for 'com.timer.chronologger' were found" UserInfo={IDEDistributionIssueSeverity=3, NSLocalizedDescription=No profiles for 'com.timer.chronologger' were found, NSLocalizedRecoverySuggestion=Xcode couldn't find any iOS App Development provisioning profiles matching 'com.timer.chronologger'.}

 EXPORT FAILED ) 
```

we would highly appreciate any help from your end as soon as possible,

ps, we have took the archive build from pipeline and exported it manually in our local mac and converted into ipa, when tried to install the ipa in iphone it said "this app cannot be installed because its integrity not be verified"

**Comment by hex2gdb** [comment]
Have tried this step:

Navigate to "Xcode—->Preferences—->Accounts." Hit the '+' icon below left to add your Apple ID

**Reply by syedimran18** [reply]
This is CI/CD Build environment we are talking about

**Reply by hex2gdb** [reply]
you might also want to read this, 

https://learn.microsoft.com/en-us/azure/devops/pipelines/apps/mobile/app-signing?view=azure-devops&tabs=apple-install-during-build

**Comment by hex2gdb** [comment]
if that did not resolve the issue you can try:

xcrun xcodebuild -exportArchive \
	...
	-allowProvisioningUpdates \
	-authenticationKeyID $(APPLE_ID_KEY_ID) \
	-authenticationKeyIssuerID $(APPLE_ID_ISSUER_ID)

**Reply by syedimranFB** [reply]
can you elaborate -authenticationKeyID & -authenticationKeyIssuerID for what they are pointing to?

**Reply by hex2gdb** [reply]
$(APPLE_ID_KEY_ID) $(APPLE_ID_ISSUER_ID) have you got this?

**Reply by hex2gdb** [reply]
and this too.

https://medium.com/@bitrise/xcode-9-code-signing-on-bitrise-f7fe31982308
